### PR TITLE
qc::is mobile_number updates

### PR DESCRIPTION
**Pull request by priyankjalu89**
image

### CODE REVIEW

**Comment by priyankjalu89** [comment]
Test Cases
-------------

- [ ] 071xxxxxxxx -> true
- [ ] 072xxxxxxxx -> true
- [ ] 073xxxxxxxx -> true
- [ ] 074xxxxxxxx -> true
- [ ] 075xxxxxxxx -> true
- [ ] 077xxxxxxxx -> true
- [ ] 078xxxxxxxx -> true
- [ ] 079xxxxxxxx -> true
- [ ] 07624xxxxxx -> true
- [ ] 07634xxxxxx -> false
- [ ] 081xxxxxxxx -> false
- [ ]

**Comment by priyankjalu89** [comment]
Ready for review.

File: tcl/is.tcl [hunk 1]
```diff
@@ -586,7 +586,7 @@ namespace eval qc::is {
     proc mobile_number {string} {
         #| Checks if the given string is of the form of a UK mobile telephone number.
         regsub -all {[^0-9]} $string {} tel_no
-        if {  [regexp {^07(5|7|8|9)[0-9]{8}$} $tel_no] } {
+        if {  [regexp {(^07(1|2|3|4|5|7|8|9)[0-9]{8}$)|(^07624[0-9]{6}$)} $tel_no] } {
```
Comment (daniel-qcode) [inline tcl/is.tcl]: the following syntax would be a little bit more concise:
```^(07[1-57-9][0-9]{8}|07624[0-9]{6})$```

Comment (priyankjalu89) [reply tcl/is.tcl]: Updated

**Comment by daniel-qcode** [comment]
Please run existing TCL tests and confirm tests pass?

**Pull request merged**